### PR TITLE
[CPU] API2.0 input issues

### PR DESCRIPTION
**Pull request by alexey-varyzgin**
### Details:
 1. *Swap Convert Transpose was made on a model input*
 2. *Ignoring Snippet after Input -> Transpose -> Convert*
 3. *Drop Convert I8->FP32 node before ScaleShift*

### Tickets:
 - *74974*
 - *77460*
 - *77417*
 - *76867*

### CODE REVIEW

File: src/plugins/intel_cpu/src/mkldnn_graph_optimizer.cpp [hunk 1]
```diff
@@ -451,6 +455,74 @@ void MKLDNNGraphOptimizer::FuseMultiplyAndAdd(MKLDNNGraph &graph) {
...
+
+    auto isSuitableParentNode = [](MKLDNNNodePtr node) {
+        return node->getType() == Convert && node->getChildEdges().size() == 1 &&
+               (node->getOriginalInputPrecisionAtPort(0) == Precision::U8 ||
+                node->getOriginalInputPrecisionAtPort(0) == Precision::I8) &&
+               node->getOriginalOutputPrecisionAtPort(0) == Precision::FP32;
+    };
+
+    auto isSuitableChildNode = [](MKLDNNNodePtr node) {
+        return node->getType() == Eltwise && (node->getAlgorithm() == EltwiseMulAdd ||
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/mkldnn_graph_optimizer.cpp:469]: Why we limit supported algorithms? Eltwise node supports U8/I8 on input for any algorithm.

File: src/plugins/intel_cpu/src/mkldnn_plugin.cpp [hunk 2]
```diff
@@ -226,6 +227,7 @@ static void TransformationUpToCPUSpecificOpSet(const std::shared_ptr<ngraph::Fun
     manager.register_pass<ngraph::pass::Validate>();
     manager.register_pass<ngraph::pass::ConvertPrecision>(precisions);
     manager.register_pass<ngraph::pass::EliminateConvert>();
+    manager.register_pass<SwapConvertTranspose>();
```
[the remark this answers is not in the document]
Comment (alexey-varyzgin) [reply src/plugins/intel_cpu/src/mkldnn_plugin.cpp:230]: SwapConvertTranspose shoud be called before MoveEltwiseUpThroughDataMov. But ConvertToCPUSpecificOpset calls after the MoveEltwiseUpThroughDataMov.

File: src/plugins/intel_cpu/src/ngraph_transformations/snippets_mark_skipped.cpp [hunk 3]
```diff
@@ -339,6 +339,9 @@ bool SnippetsMarkSkipped::run_on_model(const std::shared_ptr<ov::Model> &m) {
                     PropagateIfHasOnlyChild(node, updatedChainType);
             } else if (fusingChainType == NodeFusingType::IgnoredAfterInputs && snippets::pass::AppropriateForSubgraph(node)) {
                 SetNodeFusingType(node, NodeFusingType::IgnoredAfterInputs);
+            } else if (fusingChainType == NodeFusingType::IgnoredAfterInputs &&
+                       (ov::is_type<ngraph::op::Convert>(node) || ov::is_type<ngraph::op::v1::Transpose>(node))) {
+                SetNodeFusingType(node, NodeFusingType::IgnoredAfterInputs);
```
[the remark this answers is not in the document]
Comment (alexey-varyzgin) [reply src/plugins/intel_cpu/src/ngraph_transformations/snippets_mark_skipped.cpp]: Done

File: src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        auto convert = transpose->get_input_node_shared_ptr(0);
+        if (convert != nullptr && ov::is_type<ngraph::op::v0::Convert>(convert)) {
+            if (!(convert->get_input_element_type(0) == ngraph::element::i8 ||
+                convert->get_input_element_type(0) == ngraph::element::u8) ||
+                convert->get_output_element_type(0) != ngraph::element::f32) {
+                return false;
+            }
+            auto input = convert->get_input_node_shared_ptr(0);
+            if (input != nullptr && ngraph::op::is_parameter(input)) {
+                ngraph::replace_output_update_name(transpose->output(0), transpose->input_value(0));
```
[the remark this answers is not in the document]
Comment (alexey-varyzgin) [reply src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp]: Done

File: src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+                ngraph::copy_runtime_info(transpose, newTranspose);
+                newTranspose->set_friendly_name(transpose->get_friendly_name());
+
+                ngraph::OutputVector convertInputs = convert->input_values();
+                convertInputs[0] = newTranspose;
+                auto newConvert = convert->clone_with_new_inputs(convertInputs);
+                ngraph::copy_runtime_info(convert, newConvert);
+                newConvert->set_friendly_name(convert->get_friendly_name());
+
+                ngraph::replace_node(convert, newConvert);
```
[the remark this answers is not in the document]
Comment (alexey-varyzgin) [reply src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp]: Done

File: src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    auto transpose = ngraph::pattern::wrap_type<ngraph::op::v1::Transpose>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto transpose = std::dynamic_pointer_cast<ngraph::op::v1::Transpose>(m.get_match_root());
+
+        // Swap
+        // Input -> Convert -> Transpose
+        // to
+        // Input -> Transpose -> Convert
+        auto convert = transpose->get_input_node_shared_ptr(0);
```
[the remark this answers is not in the document]
Comment (alexey-varyzgin) [reply src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp]: Done

File: src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        auto convert = pattern_map.at(convert_m).get_node_shared_ptr();
+        auto transpose = pattern_map.at(transpose_m).get_node_shared_ptr();
+        if (convert != nullptr && ov::is_type<ngraph::opset1::Convert>(convert)) {
+            if (!(convert->get_input_element_type(0) == ngraph::element::i8 ||
+                convert->get_input_element_type(0) == ngraph::element::u8) ||
+                convert->get_output_element_type(0) != ngraph::element::f32) {
+                return false;
+            }
+            auto input = convert->get_input_node_shared_ptr(0);
+            if (input != nullptr && ngraph::op::is_parameter(input)) {
```
Comment (v-Golubev) [inline src/plugins/intel_cpu/src/ngraph_transformations/swap_convert_transpose.cpp:34]: We don't need to check precisions and nodes on nullptr here. We have already checked it in the matcher. I propose to remove it.
```suggestion
    auto input = pattern_map.at(input_m).get_node_shared_ptr();
```

**Review by v-Golubev — APPROVED** [review]
LGTM, thank you

**Comment by dmitry-gorokhov** [comment]
@IvanNovoselov to review snippets relates changes

**Review by dmitry-gorokhov — APPROVED** [review]
Can be merged after internal validation

**Comment by myshevts** [comment]
@alexey-varyzgin , @dmitry-gorokhov why this PR is not merged

**Pull request merged**